### PR TITLE
Refactor `FStringParser`

### PR DESCRIPTION
**Pull request by harupy**
Refactors `FStringParser` to simplify https://github.com/RustPython/RustPython/pull/4343.

### CODE REVIEW

File: compiler/parser/src/fstring.rs [hunk 1]
```diff
@@ -194,21 +202,21 @@ impl FStringParser {
...
+                    loop {
+                        match self.next_char() {
+                            Some(c) => {
+                                expression.push(c);
+                                if c == ch {
+                                    break;
+                                }
+                            }
+                            None => return Err(UnterminatedString),
                         }
```
Comment (burrbull) [inline compiler/parser/src/fstring.rs:218]:
```suggestion
                        let Some(c) = self.next_char() else {
                            return Err(UnterminatedString);
                        };
                        expression.push(c);
                        if c == ch {
                            break;
                        }
```

Comment (fanninpm) [reply compiler/parser/src/fstring.rs]: I keep forgetting that let-else is a thing now.

Comment (harupy) [reply compiler/parser/src/fstring.rs]: @burrbull Thanks for the suggestion!

**Review by burrbull — APPROVED** [review]

File: compiler/parser/src/fstring.rs [hunk 2]
```diff
@@ -6,63 +6,72 @@ use crate::{
...
+        }
+    }
+
+    fn next_char(&mut self) -> Option<char> {
+        self.chars.next()
+    }
+
+    fn peek(&mut self) -> Option<&char> {
+        self.chars.peek()
     }
```
Comment (fanninpm) [inline compiler/parser/src/fstring.rs:30]: @DimitrisJim @youknowone Probably not necessary, but should these two functions be `#[inline]`?

Comment (DimitrisJim) [reply compiler/parser/src/fstring.rs:30]: I'd say so, afaik the compiler usually inlines small functions like these but that isn't guaranteed, hinting it with `inline` is helps.

**Review by youknowone — APPROVED** [review]

File: compiler/parser/src/fstring.rs (hunk 2 above)
Comment (youknowone) [reply compiler/parser/src/fstring.rs:30]: I'd like to have inline for tiny functions.

**Review by youknowone — APPROVED** [review]

**Pull request merged**